lib/internal/bootstrap/node.js

```diff
@@ -613,8 +613,10 @@
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
 
-      if (exceptionHandlerState.captureFn !== null) {
-        exceptionHandlerState.captureFn(er);
+      if (exceptionHandlerState.captureFns.size !== 0) {
+        for (const fn of exceptionHandlerState.captureFns.values()) {
```

Comment:
the branch into multiple ownership here feels... off. these kinds of handlers should be at an application level.

Reply:
@devsnek Did you read the motivation for these changes in https://github.com/nodejs/node/issues/23348? If so, what alternative would you suggest to achieve the goals presented there?

Reply:
@misterdjules if we had some sort of [stopPropagation](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation) behaviour, i think this would be doable.

Reply:
I agree with @devsnek – the purpose of these hooks is to (possibly?) take ownership for the errors, otherwise we’d basically end up with a list of `uncaughtException` listeners again

Reply:
The purpose of calling all (possibly many) registered callbacks was to address the use case tested by https://github.com/misterdjules/domaine/blob/master/test/test-domaine-domain-compat.js.

Basically, if one module runs a function within a user-land implemented domain-like module and that function runs another function within another domain-like instance, when an error is thrown in that second function, the goal would be to have both domain-like instances to handle that same error. The rationale for this is that we'd want those different domain-like implementations to not step on each other by handling errors for each other (think for instance of the case where those domain-like instances are created by different dependencies of the same Node app).

However, this is an opinionated design decision and I think I'd be totally fine with only allowing the top-level active domain-like instance (`d2` in the [example mentioned above](https://github.com/misterdjules/domaine/blob/master/test/test-domaine-domain-compat.js)) to handle uncaught errors.

In that case though it seems we'd need to be able to keep a single stack of all different domain-like instances and that exposing an API to set the capture callback to user-land is not useful, as there would be a need for setting one of them internally in core. @devsnek @addaleax What are your thoughts on this?